R/utils.R

```diff
@@ -8,14 +8,39 @@
 #' @usage lhs \%>\% rhs
 NULL
 
-at_selection <- function(nm, .at){
-  if (is_quosures(.at)) {
-    lifecycle::deprecate_warn("0.4.0", I("using `vars()` in .at"))
+at_selection <- function(x, at, error_arg = caller_arg(at), error_call = caller_env()) {
+  if (is_formula(at)) {
+    at <- rlang::as_function(at, arg = error_arg, call = error_call)
+  }
+  if (is.function(at)) {
+    at <- at(names(x))
+  }
 
+  if (is.numeric(at) || is.logical(at) ||is.character(at)) {
+    if (is.character(at) && !is.null(names(x))) {
+      at <- intersect(at, names(x))
+    }
+
+    at <- vec_as_subscript(at, arg = "at", call = error_call)
```

Comment:
I don't _think_ you need to call `vec_as_subscript()` if you follow up with `vec_as_location()`, right @lionel- ?

Reply:
Oh yeah, it doesn't seem to do anything. I think it was a vestige from an experiment where I made `logical` vectors an error.

Reply:
You only need to call `vec_as_subscript()` if you use different parameters than the defaults for `vec_as_location()`.